io/zenoh-transport/tests/unicast_transport.rs

```diff
@@ -839,5 +872,398 @@ tOzot3pwe+3SJtpk90xAQrABEO0Zh2unrC8i83ySfg==
+        ]
+        .iter()
+        .map(|(k, v)| ((*k).to_owned(), (*v).to_owned())),
+    );
+    // Define the reliability and congestion control
+    let channel = [
+        Channel {
+            priority: Priority::default(),
+            reliability: Reliability::Reliable,
+        },
+        Channel {
+            priority: Priority::default(),
+            reliability: Reliability::BestEffort,
+        },
+        Channel {
+            priority: Priority::RealTime,
+            reliability: Reliability::Reliable,
+        },
+        Channel {
+            priority: Priority::RealTime,
+            reliability: Reliability::BestEffort,
+        },
+    ];
+    // Run
+    let endpoints = vec![client_endpoint];
+    let result = task::block_on(run(
+        &endpoints,
+        Some(&server_endpoint),
+        &channel,
+        &MSG_SIZE_ALL,
+    ));
+    assert!(result.is_ok())
 }
+
+#[cfg(all(feature = "transport_tls", target_family = "unix"))]
+#[test]
+fn transport_unicast_tls_two_way_auth_missing_certs_fail() {
+    use zenoh_link::tls::config::*;
+
+    task::block_on(async {
+        zasync_executor_init!();
+    });
+
+    let client_auth = "true";
+
+    // Define the locator
+    let mut client_endpoint: EndPoint = ("tls/localhost:10462").parse().unwrap();
+    client_endpoint.extend_configuration(
+        [(TLS_ROOT_CA_CERTIFICATE_RAW, CLIENT_CA)]
+            .iter()
+            .map(|(k, v)| ((*k).to_owned(), (*v).to_owned())),
+    );
+
+    // Define the locator
+    let mut server_endpoint: EndPoint = ("tls/localhost:10462").parse().unwrap();
+    server_endpoint.extend_configuration(
+        [
+            (TLS_ROOT_CA_CERTIFICATE_RAW, SERVER_CA),
+            (TLS_SERVER_CERTIFICATE_RAW, SERVER_CERT),
+            (TLS_SERVER_PRIVATE_KEY_RAW, SERVER_KEY),
+            (TLS_CLIENT_AUTH, client_auth),
+        ]
+        .iter()
+        .map(|(k, v)| ((*k).to_owned(), (*v).to_owned())),
+    );
+    // Define the reliability and congestion control
+    let channel = [
+        Channel {
+            priority: Priority::default(),
+            reliability: Reliability::Reliable,
+        },
+        Channel {
+            priority: Priority::default(),
+            reliability: Reliability::BestEffort,
+        },
+        Channel {
+            priority: Priority::RealTime,
+            reliability: Reliability::Reliable,
+        },
+        Channel {
+            priority: Priority::RealTime,
+            reliability: Reliability::BestEffort,
+        },
+    ];
+    // Run
+    let endpoints = vec![client_endpoint];
+    let result = task::block_on(run(
+        &endpoints,
+        Some(&server_endpoint),
+        &channel,
+        &MSG_SIZE_ALL,
+    ));
+    assert!(result.is_err());
+    let err = result.unwrap_err();
+    assert!(err
+        .to_string()
+        .contains(RUSTLS_CERTIFICATE_REQUIRED_ALERT_DESCRIPTION));
+}
+
+#[cfg(all(feature = "transport_tls", target_family = "unix"))]
```

Comment:
I've just spotted this (also in the old `tls` test): should we try to remove the `target_family` to also target `windows` environments?

Reply:
Ok, let's keep it then :)